ocaml/xapi/test_vdi_cbt.ml

```diff
@@ -218,3 +243,65 @@ let test =
   ]
+
+(* Confirm VDI.data_destroy changes requisite fields of VDI *)
+let test_vdi_after_data_destroy () =
+  let __context = Test_common.make_test_database () in
+  let sR,vDI = make_mock_server_infrastructure ~__context in
+  Db.VDI.set_type ~__context ~self:vDI ~value:`suspend;
+  let vM = Test_common.make_vm ~__context () in
+  let vBD = Test_common.make_vbd ~__context ~uuid:"VBD-1" ~vDI ~vM ~currently_attached:false () in
+  let _: _ API.Ref.t = Test_common.make_vbd ~__context ~uuid:"VBD-1" ~vDI ~vM ~currently_attached:false () in
+
+  let check_vdi_is_snapshot_and_type ~vDI ~snapshot ~vdi_type ~managed =
+    let open Printf in
+    OUnit.assert_equal ~msg:(sprintf "VDI type should be set to %s" (Record_util.vdi_type_to_string vdi_type))
+      (Db.VDI.get_type ~__context ~self:vDI) vdi_type;
+    OUnit.assert_equal ~msg:(sprintf "VDI managed should be set to %b" managed)
+      (Db.VDI.get_managed ~__context ~self:vDI) managed;
+    let word = if snapshot then "" else " not" in
+    OUnit.assert_equal ~msg:(sprintf "VDI should%s be a snapshot" word)
+      (Db.VDI.get_is_a_snapshot ~__context ~self:vDI) snapshot
+  in
+  check_vdi_is_snapshot_and_type ~vDI ~snapshot:true ~vdi_type:`suspend ~managed:true;
+
+  (* set vDI as the suspend VDI of vM *)
+  Db.VM.set_suspend_VDI ~__context ~self:vM ~value:vDI;
+  OUnit.assert_equal ~msg:"VM.suspend_VDI should point to previously created VDI"
+    (Db.VM.get_suspend_VDI ~__context ~self:vM) vDI;
+
+  OUnit.assert_equal ~msg:"VDI should link to previously created VBD"
+   (Db.VDI.get_VBDs ~__context ~self:vDI) [vBD];
+
+  (* run VDI.data_destroy, check it has updated VDI fields *)
+  Xapi_vdi.data_destroy ~__context ~self:vDI;
+
+  OUnit.assert_equal ~msg:"VDI.data_destroy should set VDI type to cbt_metadata"
+    (Db.VDI.get_type ~__context ~self:vDI) `cbt_metadata;
+
+  OUnit.assert_equal ~msg:"VDI.data_destroy should destroy all associated VBDs"
+    (Db.VDI.get_VBDs ~__context ~self:vDI) [];
+
+  OUnit.assert_equal ~msg:"VM.suspend_VDI should be set to null"
+    (Db.VM.get_suspend_VDI ~__context ~self:vM) Ref.null;
+
+  (* check for idempotence for metadata snapshot VDIs *)
+  check_vdi_is_snapshot_and_type ~vDI ~snapshot:true ~vdi_type:`cbt_metadata ~managed:true;
+  Xapi_vdi.data_destroy ~__context ~self:vDI;
+  check_vdi_is_snapshot_and_type ~vDI ~snapshot:true ~vdi_type:`cbt_metadata ~managed:true
+
+(* check VDI.data_destroy throws VDI_NOT_MANAGED if managed:false *)
+let test_vdi_managed_data_destroy () =
+  let __context = Test_common.make_test_database () in
+  let _,vDI = make_mock_server_infrastructure ~__context in
+  Db.VDI.set_managed ~__context ~self:vDI ~value:false;
+  OUnit.assert_raises ~msg:"VDI.data_destroy only works on managed VDI"
+    Api_errors.(Server_error (vdi_not_managed, [Ref.string_of vDI]))
+    (fun () -> Xapi_vdi.data_destroy ~__context ~self:vDI)
+
+let test_vdi_data_destroy =
```

Comment:
oops, forgot one thing: this has to be above `let tests =`, and `test_vdi_data_destroy` should be added to it to make sure it's run.

Reply:
You can also bring the above two functions into the body of this let binding, to make things more local (see `test_vdi_allowed_operations.ml` for an example).